lib/phoenix_live_view/test/live_view_test.ex

```diff
@@ -920,6 +922,64 @@ defmodule Phoenix.LiveViewTest do
     %Element{proxy: proxy, selector: selector, form_data: form_data}
   end
 
+  @doc """
+  Builds a file input for testing uploads within a form.
+
+  Given the form DOM ID, the upload name, and a map of client metadata
+  for the upload, the returned file input can be passed to `render_upload/2`.
+
+  Client metadata takes the following form:
+
+    * `:last_modified` - the last modified timestamp
+    * `:name` - the name of the file
+    * `:content` - the binary content of the file
+    * `:size` - the byte size of the content
+    * `:type` - the MIME type of the file
+
+  ## Examples
+
+      avatar = file_input(lv, "my-form-id", :avatar, %{
+        last_modified: 1_594_171_879_000,
+        name: "myfile.jpeg",
+        content: File.read!("myfile.jpg"),
+        size: 1_396_009,
+        type: "image/jpeg"
+      })
+
+      assert render_upload(avatar, "foo.jpeg") =~ "100%"
+  """
+  defmacro file_input(view, form_selector, name, entries) do
```

Comment:
Why does this needs to be a macro? And if it has to be a macro, we should move the bulk of `quote` to a `__file_input__` function.

Reply:
It needs to be a macro because `ChannelTest.connect/3`, which we have to invoke locally so I'm not sure we can delegate out much more than currently